client/app/vps/vps-task.service.js

```diff
@@ -0,0 +1,138 @@
+class VpsTaskService {
+    constructor ($http, $q, $rootScope, $timeout, $translate, CloudMessage) {
+        this.$http = $http;
+        this.$q = $q;
+        this.$rootScope = $rootScope;
+
+        this.$timeout = $timeout;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+
+        this.firstCall = true;
+    }
+
+    /*
+     * imported from vps services to avoid calling this services
+     * + getSelectedVps
+     * + getTaskInPropgress
+     */
+    getSelectedVps (serviceName) {
+        return this.$http.get(["/sws/vps", serviceName,"info"].join("/"), {serviceType: "aapi"})
+            .then(result => result.data)
+            .catch(err => this.$q.reject(err));
+    }
+
+    getTaskInProgress (serviceName, type) {
+        var result = null;
+        return this.getSelectedVps(serviceName).then(vps => {
+            if (vps && vps.name) {
+                return this.$http.get(["/sws/vps", vps.name, "tasks/uncompleted"].join("/"), {
```

Comment:
this could be done in a separate function or service

Reply:
Done